samples/web_shell/scripts/git.js

```diff
@@ -0,0 +1,319 @@
+// Copyright 2015 Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Run a local proxy server to inject CORS headers.
+// TODO(grv): Host a live server on app engine.
+var PROXY_SERVER_URL = 'pepper_http://localhost:8080/';
+var GIT_NMF_FILE_PATH = 'scripts/resources/git';
+
+// The subject field sent with each message to identify it.
+var messageId = 1;
+
+document.currentScript.ready(function(cx) {
+  var GIT_CMD_USAGE_STRING = 'usage: git [options]';
+
+  /**
+   * @return {void}
+   */
+  var gitMain = function(cx) {
+    cx.ready();
+
+    /** @type {Array<string>} */
+    var list = cx.getArg('_', []);
+    if (list.length != 3 || cx.getArg('help')) {
+      cx.stdout.write(GIT_CMD_USAGE_STRING + '\n');
+      return cx.closeOk();
+    }
+
+    createMessage(cx).then(function(message) {
```

Comment:
It would be nicer to wrap all this code into a separate function, so that we don't have to hunt for the `cx.closeOk` call. Something along the line of:

```
processGitCommand(cx).then(function() {
  cx.closeOk();
}).catch(function(error)) {
  cx.closeError(error);
});
```